demo/docker/nginx/nginx.conf

```diff
@@ -24,8 +25,15 @@ http {
 #      proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
 #    }
 
+    # this is the internal Docker DNS, cache only for 30s
+    # This setting + substitution pattern used below is because we want ot have separated compose for app and environment.
+    # More details you can find here: https://sandro-keil.de/blog/let-nginx-start-if-upstream-host-is-unavailable-or-down/
+    # It not works with services (e.g.) grafana (error: "Grafana has failed to load its application files")
+    resolver 127.0.0.11 valid=30s;
```

Comment:
Mayby move this at the start of `server`

Reply:
The problem is here:
> It not works with services (e.g.) grafana (error: "Grafana has failed to load its application files")
Firstly I've tried to do this trick with all services but it cause some problems with grafana. I don't want to someone walk through the same way ;-)